testbed/addons/karpenter/construct.ts

```diff
@@ -4,24 +4,25 @@ import iam = require('@aws-cdk/aws-iam');
 
 export interface KarpenterProps {
     cluster: eks.Cluster;
-    namespace?: string;
 }
 
 export class Karpenter extends cdk.Construct {
+    private namespace: string;
     constructor(scope: cdk.Construct, id: string, props: KarpenterProps) {
         super(scope, id);
-
-        const ns = props.cluster.addManifest('namespace', {
-            apiVersion: 'v1',
-            kind: 'Namespace',
-            metadata: {
-                name: props.namespace ?? "karpenter"
+        this.namespace = "karpenter"
```

Comment:
instead, consider props.namespace = props.namespace ?? "karpenter", and then just use that existing field. There might even be an operator `props.namespace ?= "karpenter"`